src/lib.rs

```diff
@@ -1,15 +1,15 @@
 //! The `Ipld` crate.
 
-#![deny(missing_docs)]
+//#![deny(missing_docs)]
```

Comment:
What about changing this to `#![warn(missing_docs)]` or something that doesn't break CI but also doesn't just eschew code-level docs altogether?

I can see that this PR is substantial and I appreciate that not every single function and trait tediously needs a comment like `// Decodes CBOR` but I'd rather err on the side of over-documenting than under-documenting.